src/main/webapp/app/core/language/language.helper.ts

```diff
@@ -53,8 +57,13 @@ export class JhiLanguageHelper {
         });
     }
 
-    private getPageTitle(routeSnapshot: ActivatedRouteSnapshot) {
-        let title: string = routeSnapshot.data && routeSnapshot.data['pageTitle'] ? routeSnapshot.data['pageTitle'] : 'artemisApp';
+    /**
+     * Get the current page's title key based on the router state.
+     * Fallback to 'global.title' when no key is found.
+     * @param routeSnapshot The snapshot of the current route
+     */
+    getPageTitle(routeSnapshot: ActivatedRouteSnapshot) {
+        let title: string = routeSnapshot.data && routeSnapshot.data['pageTitle'] ? routeSnapshot.data['pageTitle'] : 'global.title';
```

Comment:
As far as I know ?? is preferred over || for default values, since || would override 0 or an empty string.
But overriding an empty string might actually be the preferred behavior here

Reply:
Good spot, even shorter: `routeSnapshot.data?.['pageTitle'] || 'global.title'`.